### ISSUE TITLE
Error during HostedPaymentPages request

### ISSUE THREAD

**Issue by san4esmc** [Adyen/adyen-java-api-library#552]
**Library version**: 15.0.0
**Description**
During refactoring done in this commit:
https://github.com/Adyen/adyen-java-api-library/commit/cbc2e3a53a3aedd733a1b16f27ae737df496a110#diff-f65e38ffa37baaa561776830e7ed9049ddc7a50e98a377f16cda9a1bbe03b98d
the part to check POST body was removed and now when you try to execute HostedPaymentPages request it passes null inside constructor for StringEntity, which leads to exception.

The fix would be to introduce back a check and replace AdyenHttpClient line 167:
```
httpPost.setEntity(new StringEntity(requestBody));
```
with
```
StringEntity requestEntity = null;
if (requestBody != null && !requestBody.isEmpty()) {
    requestEntity = new StringEntity(requestBody); 
}
httpPost.setEntity(requestEntity);
```

**Comment by san4esmc** [comment]
But the provided fix will only fix the issue with POST requests and I see that PATCH requests have same issue, so probably check should be done before switch

**Comment by Aleffio** [comment]
Hi @san4esmc, thanks for opening the issue,
we are investigating on this, and will provide a fix.
Have you considered using `/paymentMethods` instead? https://docs.adyen.com/online-payments/api-only#step-1-get-available-payment-methods, the page also contains a snippet using this Java library.

Regards,
Alessio
Adyen

**Comment by san4esmc** [comment]
Hi, @Aleffio !
Yes, we are using `/paymentMethods` also, but not all payment methods were migrated to web components yet, so we have a mixed setup with older API + new API and we are really dependant now on HostedPaymentPages request.
Hope it will be fixed soon!
Thanks

**Comment by achawki** [comment]
Hi @san4esmc,
thanks again for opening the issue. 
It should be fixed with the release  [15.0.2](https://github.com/Adyen/adyen-java-api-library/releases/tag/15.0.2)
